lib/spree/core/mail_settings.rb

```diff
@@ -12,7 +12,7 @@ def self.init
       end
 
       def override!
-        ActionMailer::Base.default_url_options[:host] ||= Config.site_url
+        ActionMailer::Base.default_url_options[:host] ||= ENV.fetch("SITE_URL", Config.site_url)
```

Comment:
why not just
ActionMailer::Base.default_url_options[:host] ||= Config.site_url
?

Reply:
I think we ultimately want to move towards depending on the `ENV` vars and not on the `Spree::Config` values, but here in some cases `SITE_URL` might not be set (testing and dev for example). I was rushing a bit here to unblock the release...

I started a second branch after this one that basically deletes `Spree::Core::MailSettings` completely and merges a simplified version of it's logic into `MailConfiguration` and starts dropping a lot of these `Spree::Config` settings.